keras/preprocessing/image.py

```diff
@@ -70,9 +70,13 @@ def random_barrel_transform(x, intensity):
     # TODO
     pass
 
-def random_channel_shift(x, rg):
-    # TODO
-    pass
+def random_channel_shift(x, intensity, channel_index=0):
+    x = np.rollaxis(x, channel_index, 0)
+    channel_images = [np.clip(x_channel + np.random.uniform(-intensity, intensity), 0, 1) 
```

Comment:
Why clip to [0,1]?
There's no guarantee that X passed to the datagenerator will be within 0 and 1.